src/lib/fs-utils.ts

```diff
@@ -31,12 +35,10 @@ export function writeFileAtomic(
 export function safeRename(src: string, dest: string): Bluebird<void> {
 	return Bluebird.resolve(fs.rename(src, dest))
 		.then(() => fs.open(Path.dirname(dest), 'r'))
-		.tap(fs.fsync)
-		.then(fs.close);
+		.tap((fh) => fh.sync())
+		.then((fh) => fh.close());
 }
 
-export function pathExistsOnHost(p: string): Bluebird<boolean> {
```

Comment:
Having looked at the code I think it's worth having a helper function for `pathExistsOnHost` to make it clearer when a check is looking at the host fs and centralize the logic of how to check host fs vs container fs in a single place rather than copying that logic in multiple places where a mistake might get introduced

Reply:
Makes sense.

Reply:
I'll be working on this a bit more, will ping you again when I've pushed a proposed refactor if the refactor ends up being small enough

Reply:
I've centralized the various fs calls within the codebase to use fs-utils. Examples include: 
- Replacing any `fs.stat` calls that are for the purpose of checking file existence with `existsAsync`
- Replacing multiple unlink calls for deleting files with a new fs-utils function, unlinkAll, which takes one or more filenames as arguments. This is because everywhere unlink is being called, it needed to have a .catch to handle when the file doesn't exist. `unlinkAll` does this in one place so we can deduplicate this functionality.
- Replace multiple calls to `mkdir(dirName, { recursive: true })` with an annotated fs-utils `mkdirRecur` to be clear about this functionality
- Refactor fs-utils to use `async`/`await` and native Promises instead of Bluebird

Misc:
- Remove an unused function from config/configJson.ts

Untouched fs functionality:
- pathOnHost() in config/configJson.ts, as it has extra context-specific calls that would make replacing it with a generic fs-util function unnecessarily convoluted.
- remountAndWriteAtomic() in config/backend.ts, as it's used only in the context of the device backends and probably won't see use anywhere else.

What I interpreted from your comment @Page- was the idea of centralizing various vague or multi-step fs functionalities, such as fs.unlink needing a catch whenever it's called. AFAIK there are no uses for `pathExistsOnHost` to deal with container FS's. I have added a comment under pathExistsOnHost in an attempt to clarify it.